pkg/controller/node/controller_utils.go

```diff
@@ -284,7 +323,7 @@ func recordNodeStatusChange(recorder record.EventRecorder, node *api.Node, new_s
 // terminatePods will ensure all pods on the given node that are in terminating state are eventually
 // cleaned up. Returns true if the node has no pods in terminating state, a duration that indicates how
 // long before we should check again (the next deadline for a pod to complete), or an error.
-func terminatePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName string, since time.Time, maxGracePeriod time.Duration) (bool, time.Duration, error) {
+func terminatePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName string, nodeStatus nodeStatusData, since time.Time, maxGracePeriod time.Duration) (bool, time.Duration, error) {
```

Comment:
I don't think you need to change this function. Forgiveness should not affect what happens when a pod is deleted. It should only delay making the request to delete the pod (due to the node becoming not ready).